src-goocanvas2/gooCanvas.ml

```diff
@@ -0,0 +1,228 @@
+      Canvas.convert_to_item_space obj i#as_item x y
+    method convert_from_item_space (i:base_item) ~x ~y =
+      Canvas.convert_from_item_space obj i#as_item x y
+    method convert_to_pixels ~x ~y = Canvas.convert_to_pixels obj x y
+    method convert_from_pixels ~x ~y = Canvas.convert_from_pixels obj x y
+    method get_scale = Canvas.get_scale obj
+    method set_scale s = Canvas.set_scale obj s
+    method get_item_at ?(is_pointer_event=false) ~x ~y () =
+      match Canvas.get_item_at obj x y is_pointer_event with
+      | None -> None
+      | Some obj -> let o = new item obj in Some (o:>base_item)
+    method scroll_to ~x ~y = Canvas.scroll_to obj x y
+  end
+
+let canvas =
+  GContainer.pack_container [] ~create:(fun pl ->
+    let w = Canvas.new_canvas () in
+    Gobject.set_params w pl;
+    new canvas w)
+
+
+class text (obj : GooCanvas_types.text obj) =
+  object(self)
+    inherit [text_p] item obj
+  end
+
+let text ?(text="") ?(x=0.) ?(y=0.) ?(width=(-1.)) ?(props=[]) parent =
+  let t = new text (Text.new_text parent#as_item text ~x ~y ~width) in
+  t#set props ;
+  t
+
+class rect (obj : GooCanvas_types.rect obj) =
+  object(self)
+    inherit [rect_p] item obj
+  end
+
+let rect ?(x=0.) ?(y=0.) ?(width=(-1.)) ?(height=(-1.)) ?(props=[]) parent =
+  let t = new rect (Rect.new_rect parent#as_item ~x ~y ~width ~height) in
+  t#set props ;
+  t
+
+class widget (obj : GooCanvas_types.widget obj) =
+  object(self)
+    inherit [widget_p] item obj
+  end
+
+let widget ?(x=0.) ?(y=0.) ?(width=(-1.)) ?(height=(-1.)) ?(props=[]) w parent =
+  let t = new widget (GtkGooCanvas.Widget.new_widget parent#as_item w ~x ~y ~width ~height) in
+  t#set props ;
+  t
+
+class group (obj : GooCanvas_types.group obj) =
+  object(self)
+    inherit [group_p] item obj
+    method width = Gobject.Property.get obj GtkGooCanvas.Group.width
+  end
+
+let group ?(props=[]) ?parent () =
+  let parent = match parent with None -> None | Some p -> Some p#as_item in
+  let t = new group (Group.new_group parent) in
+  t#set props ;
+  t
+
+class polyline (obj : GooCanvas_types.polyline obj) =
+  object(self)
+    inherit [polyline_p] item obj
+  end
+
+let canvas_points =
+  let from_pairs =
+    let rec iter acc = function
+    | [] -> List.rev acc
+    | (x, y) :: q -> iter (y :: x :: acc) q
+    in
+    iter []
+  in
+  fun l ->
+    let t = Array.of_list (from_pairs l) in
+    let len = Array.length t in
+    GtkGooCanvas.Points.new_points len t
+
+let polyline ?(close_path=true) ?points ?(props=[]) parent =
+  let t = new polyline (Polyline.new_polyline parent#as_item close_path) in
+  let props =
+    match points with
+    | None -> props
+    | Some l -> `POINTS (canvas_points l) :: props
+  in
+  t#set props ;
+  t
+
+let polyline_line ~x1 ~y1 ~x2 ~y2 ?(props=[]) parent =
+  let t = new polyline (Polyline.new_polyline_line parent#as_item
+    x1 y1 x2 y2)
+  in
+  t#set props ;
+  t
+
+type pixbuf
```

Comment:
What is this `pixbuf` type, that doesn't seem to be used anywhere?

Reply:
Indeed, that's an error. I fix it in the next commit.